### PR TITLE
Core/Ulduar: Multiple fixes to Flame Leviathan

### LINKED ISSUE #351: 10 Man Raid - Ulduar - Flame Leviathan Shutdown [$20]
Description:
Flame Leviathan doesn't go into Shutdown when turrets are destroyed.

Current behaviour: Tell us what happens
If a player is launched onto FL he/she can target and destroy the turrets on FL as expected. However, destorying these turrets currently causes no effect,

Expected behaviour: Tell us what should happen instead
Destroying the turrets should cause FL to become stunned, increase its damage taken by 50% and reset its gather speed buff.

Steps to reproduce the problem:

Engage FL.
Launch player onto FL.
Player destroys turrets.
http://wowwiki.wikia.com/wiki/Flame_Leviathan

Branch(es): 0.x / 1.x / master (Specify the branch(es) affected by this issue)

AC hash/commit:
a0ade8571fbc16f9d8452966e89faefd76fab07e
Operating system:
server 2008

Modules: Are you using modules? if yes, list them
none

Other Customizations: Are you use core/db patches or other customizations? If yes please specify them here.
none

<bountysource-plugin>

---
There is a **[$20 open bounty](https://www.bountysource.com/issues/41347365-10-man-raid-ulduar-flame-leviathan-shutdown?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** on this issue. Add to the bounty at [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Nefertumm**
##### CHANGES PROPOSED:

-  Remove almost the hardcoded text to allow translating.
-  Fix Shutdown abilitie from Flame Leviathan.
-  If someone dies while riding the Flame Leviathan, we should kick out of the vehicle.
-  Flame Leviathan now talks when a player rides him.
-  Added missing overrides and replaced NULL for nullptr.

###### ISSUES ADDRESSED:

Closes #351 
Updates #570 

##### TESTS PERFORMED:

Compiled and test all the things, all should work. 

##### HOW TO TEST THE CHANGES:

In order to test the shutdown, you will need 2 or more players or just open 2+ WoW.
Go to Ulduar, get on a vehicle, and start the fight with Flame Leviathan.

##### KNOWN ISSUES AND TODO LIST:

- [ ] Turrets should be immune to vehicle's damage.

There are some of hardcoded gossips and text left. For gossips we have to wait until #1338 gets merge.

##### Target branch(es):

Master

### CODE REVIEW

**Comment by Nefertumm** [comment]
Travis error makes no sense to me...

File: src/server/scripts/Northrend/Ulduar/Ulduar/boss_flame_leviathan.cpp [hunk 1]
```diff
@@ -675,6 +644,9 @@ void boss_flame_leviathan::boss_flame_leviathanAI::KilledUnit(Unit* who)
 {
     if (who == me->GetVictim())
         events.RescheduleEvent(EVENT_PURSUE, 0);
+
+    if (Player* plr = who->ToPlayer())
```
Comment (FrancescoBorzi) [inline src/server/scripts/Northrend/Ulduar/Ulduar/boss_flame_leviathan.cpp]: @Nefertumm here you are declaring the `plr` variable but then never using it. That's why Travis is complaining.

Comment (Nefertumm) [reply src/server/scripts/Northrend/Ulduar/Ulduar/boss_flame_leviathan.cpp]: Oh, i didn't see. I forgot to change it.

File: src/server/scripts/Northrend/Ulduar/Ulduar/boss_flame_leviathan.cpp [hunk 2]
```diff
@@ -1348,15 +1362,18 @@ class npc_lore_keeper_of_norgannon_ulduar : public CreatureScript
...
         {
             if (_eventStarted)
                 return;
 
             if (param == ACTION_START_NORGANNON_EVENT)
             {
                 if (Creature* cr = me->FindNearestCreature(NPC_HIGH_EXPLORER_DELLORAH, 20.0f, true))
+                {
+                    _dellorah = cr;
                     _dellorahGUID = cr->GetGUID();
```
Comment (GitVerge) [inline src/server/scripts/Northrend/Ulduar/Ulduar/boss_flame_leviathan.cpp]: _dellorahGUID could be removed

**Comment by comix1988** [comment]
@Nefertumm I tested this PR in dualbox:
-now the flame leviathan goes into shoutdown
schermata 2019-02-23 13 01 59
test2


-when a player rides him he talks
test1

-when a player who rides him dies he is thrown to the ground

tested on Ubuntu 18.04.2 LTS

**Comment by Nefertumm** [comment]
Not ready yet.

**Comment by Nefertumm** [comment]
I have encountered an issue with turrets, they are taking damage from vehicles when it shouldn't. Actually I'm trying to figure out how to avoid this. Seems like [this](https://github.com/azerothcore/azerothcore-wotlk/blob/15622225d642d9db207ebacf920d30c061aaa57f/src/server/scripts/Northrend/Ulduar/Ulduar/boss_flame_leviathan.cpp#L805) isn't working as intended.

**Comment by Nefertumm** [comment]
After some checks i realized that the problem is only visual. It's ready to be merged.

**Comment by Pondaveia** [comment]
Just to clear things, did the turrets get fixed?

**Comment by Nefertumm** [comment]
The turrets were fine.

**Pull request merged**